pkg/api/unversioned/types.go

```diff
@@ -33,6 +38,8 @@ type TypeMeta struct {
 	// may reject unrecognized values.
 	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#resources
 	APIVersion string `json:"apiVersion,omitempty"`
+
+	Raw *RawData
```

Comment:
I think it's kind of a non-starter to let an optimization like this leak nonsense into our API. I'm less willing to take this than even super gross things like using the unsafe package to turn a runtime.Object into a uintptr, which we use as the key to a map with encodings. (!)